homework/unique_ptr/unique_ptr.hpp

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+namespace coders {
+
+template <typename T>
+class unique_ptr{
+public:
+    unique_ptr() = delete;
+    unique_ptr(T* ptr);
+    unique_ptr(const unique_ptr<T>&) = delete;
+    unique_ptr(unique_ptr<T>&& ptr) noexcept;
+    ~unique_ptr();
+
+    T* get() const noexcept;
+    T* release() noexcept;
+    void reset(T* ptr) noexcept;
```

Comment:
```suggestion
    void reset(T* ptr = nullptr) noexcept;
```
Default argument would be useful